### PR TITLE
PLT-6847 withdraw by payout

### PR DESCRIPTION
**Pull request by jhbertra**
- [x] `Withdraw` now accepts payout tx out refs instead of contact ID + token name
- [x] all tests and clients updated
- [x] REST API updated
- [x] `Payout` schema changed to align with other entities and provide more consistent interface (thanks @nhenin)
  - `PayoutRef` renamed to `PayoutHeader`
  - Consistent use of `payoutId` vs `payout`
  - use `AssetId` for `role` field instead of two fields
  - use more clear terminology and an explicit status field for withdrawn / available payout filtering.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

File: marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/REST/ApiError.hs [hunk 1]
```diff
@@ -111,43 +110,32 @@ rangeNotSatisfiable msg errorCode = toServerError . ApiError msg (fromMaybe "Ran
...
-instance HasDTO (CreateError 'V1) where
-  type DTO (CreateError 'V1) = ApiError
-
-instance ToDTO (CreateError 'V1) where
+    WithdrawConstraintError err -> ApiError (show err) "ConstraintError" Null case err of
+      MintingUtxoNotFound _ -> 400
+      RoleTokenNotFound _ -> 400
+      PayoutNotFound _ -> 400
+      CoinSelectionFailed _ -> 400
+      _ -> 500
```
Comment (bwbush) [inline marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/REST/ApiError.hs:124]: Doesn't this change make error reporting *less* expressive? The error reports were already too terse.

File: marlowe-runtime/changelog.d/20230822_161710_jhbertra_plt_6847_withdraw_by_payout.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+### Changed
+
+- BREAKING `Withdraw` now accepts a set of payout tx outs instead of a contract
+  ID and a role token. The old behaviour can be emulated via a query to fetch
+  unclaimed payouts for a contract.
```
Comment (bwbush) [inline marlowe-runtime/changelog.d/20230822_161710_jhbertra_plt_6847_withdraw_by_payout.md:5]: We'll have to revise some starter-kit lessons prior to releasing this.

**Comment by nhenin** [comment]
@jhbertra @bwbush , I have finished the integration on the ts-sdk and added a couple of fixes at the end of your commits @jhbertra .

File: marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/REST/ApiError.hs (hunk 1 above)
Comment (jhbertra) [reply marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/REST/ApiError.hs:124]: It doesn't make it less expressive, it actually makes it more expressive:

```hs
-- before on "PayoutInputNotFound"
ApiError "Internal error" "PayoutInputNotFound" Null 500

-- after
ApiError "PayoutNotFound <payout tx in>" "ConstraintError" 500

**Pull request merged**